Modules/core/Controller/CoretilesController.php

```diff
@@ -207,6 +207,14 @@ public function showMainSubMenu($id){
             $title = $modelSubMenu->getName($id);
         }
 
+        // filter out items with no space
+        $refLength = count($items);
+        for ($i=0; $i<$refLength; $i++) {
+            if (!$items[$i]) {
+                unset($items[$i]);
```

Comment:
if you unset, next unset will be at wrong index at items has been modified meanwhile

Reply:
ah yes, doesn't handle the case where there's 2 items to unset in a row

Reply:
Because if a space is deleted, we don't need its tile anymore, unless we remember it exists and use it for the next space we create